src/neptune/metadata_containers/metadata_container.py

```diff
@@ -129,13 +152,28 @@ def __init__(
         self._workspace: str = self._api_object.workspace
         self._project_name: str = self._api_object.project_name
 
+        self._async_lag_threshold = async_lag_threshold
+        self._async_lag_callback = self._get_callback(
+            provided=async_lag_callback,
+            env_name=NEPTUNE_ENABLE_DEFAULT_ASYNC_LAG_CALLBACK,
+        )
+        self._async_no_progress_threshold = async_no_progress_threshold
+        self._async_no_progress_callback = self._get_callback(
+            provided=async_no_progress_callback,
+            env_name=NEPTUNE_ENABLE_DEFAULT_ASYNC_NO_PROGRESS_CALLBACK,
+        )
+
         self._op_processor: OperationProcessor = get_operation_processor(
             mode=mode,
             container_id=self._id,
             container_type=self.container_type,
             backend=self._backend,
             lock=self._lock,
             flush_period=flush_period,
+            async_lag_callback=self._async_lag_callback_method,
```

Comment:
```
async_lag_callback=partial(async_lag_callback, self) if async_lag_callback else None,
async_no_progress_callback=partial(async_no_progress_callback, self) if async_no_progress_callback else None,
```
?
Then we don't need those two additional methods `_async_lag_callback_method`, `_async_no_progress_callback_method`

Reply:
Done